src/printer/cvc/cvc_printer.cpp

```diff
@@ -404,7 +408,12 @@ void CvcPrinter::toStream(
     case kind::APPLY_SELECTOR_TOTAL: {
         TypeNode t = n[0].getType();
         Node opn = n.getOperator();
-        if (t.isTuple() || t.isRecord())
+        if (!t.isDatatype())
+        {
+          toStream(op, opn, depth, types, false);
+        }
+        else if (t.isTuple()
+                 || static_cast<DatatypeType>(t.toType()).isRecord())
```

Comment:
(see other comments about `static_cast<DatatypeType>`)

```suggestion
                 || DatatypeType(t.toType()).isRecord())
```